### PR TITLE
doc: add CLI reference docs - part 1

### PR DESCRIPTION
**Pull request by FeynmanZhou**
Fix: #59 , see [preview](https://deploy-preview-66--oras-project.netlify.app/cli_reference/)

Signed-off-by: Feynman Zhou <feynmanzhou@microsoft.com>

### CODE REVIEW

**Comment by qweeah** [comment]
Looks like a lot of commands are missing. like `push`, `pull` and `manifest` sub-commands. Is this intended?

File: docs/CLI_Reference/index.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  tag         Tag a manifest in the remote registry
+  version     Show the oras version information
+
+Flags:
+  -h, --help   help for oras
+
+Use "oras [command] --help" for more information about a command.
+```
+
+## See also
```
Comment (qweeah) [inline docs/CLI_Reference/index.md:32]: Hyperlink target name is not right in this section, e.g. `oras_attach.md` should be `0_oras_attach.md`

**Comment by FeynmanZhou** [comment]
> Looks like a lot of commands are missing. like `push`, `pull` and `manifest` sub-commands. Is this intended?

Yes, this is part 1. If this format and layout look good to us, I will submit the remaining commands in another PRs. This will also make PR reviewing more convenient.

**Comment by qweeah** [comment]
> > Looks like a lot of commands are missing. like `push`, `pull` and `manifest` sub-commands. Is this intended?
> 
> Yes, this is part 1. If this format and layout look good to us, I will submit the remaining commands in another PRs. This will also make PR reviewing more convenient.

The format and layout looks good.

**Review by qweeah — APPROVED** [review]
LGTM

**Review by sajayantony — APPROVED** [review]

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**